### PR TITLE
feature: Add release-drafter to project

### PR DESCRIPTION
**Pull request by jack1902**
to:
cc: @subspacecommunity/subspace-maintainers
related to:
resolves:

## Background

Adds release-drafter for consistent releases with snippets of what has recently been added based on PR titles

### Changes

* Add initial config for `release-drafter`

## Testing

I've used this on my own projects and it is great for getting a release together over time

### CODE REVIEW

File: .github/release-drafter.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      - '/fix\/.+/'
+    title:
+      - '/fix/i'
+  - label: 'enhancement'
+    branch:
+      - '/feature\/.+/'
+    body:
+      - '/feature{0,1}\/.+/'
+    title:
+      - '/feature/i'
```
Comment (gchamon) [inline .github/release-drafter.yml:35]: might be nice to add label `code-quality` or something related when pushing from `refactor/*` branches

File: .github/release-drafter.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Config for https://github.com/apps/release-drafter
+name-template: 'v$NEXT_MINOR_VERSION (🌈 Insert Release Name)'
+tag-template: 'v$NEXT_MINOR_VERSION'
+categories:
+  - title: 'Dependency Updates'
+    label: 'dependencies'
+  - title: '🚀 Features'
+    label: 'feature'
```
Comment (gchamon) [inline .github/release-drafter.yml:8]: same here

Comment (jack1902) [reply .github/release-drafter.yml:8]: Not sure what you mean by `same here` ?

Comment (gchamon) [reply .github/release-drafter.yml]: To also have support for code quality branches

**Review by gchamon — APPROVED** [review]

**Review by agonbar — APPROVED** [review]
Nice!

**Pull request merged**